### DISCUSSION TITLE
Problem with ARCH building in CI/CD (Apple M1)

### DISCUSSION THREAD

**Discussion by sdurban** [swiftyfinch/Rugby#355] [Q&A] [answered]
Hello,

I am currently integrating Rugby into our company's CI/CD pipeline using Codemagic on a Mac mini M1. However, we've encountered an issue where it seems to be looking for combined binaries (or something along those lines), leading to a compilation error. Here are the warnings we receive during the build process (we're just trying to run tests in a simulator):

> ld: warning: search path '/Users/builder/.rugby/bin/Adjust/ReleaseDev-iphonesimulator-arm64 x86_64/33ca3ee' not found
> ld: warning: search path '/Users/builder/.rugby/bin/Amplitude/ReleaseDev-iphonesimulator-arm64 x86_64/a176cae' not found

The system appears to be searching for combined amd64 and x86_64 binaries. This issue does not occur on our local development machines, which leads me to believe there might be a discrepancy or a missing configuration within XCode that we are not aware of.

Could these paths be generated, or is there a specific XCode configuration that we're missing? Any guidance or advice on how to resolve this would be greatly appreciated.

Thank you!

**Comment by swiftyfinch** [comment]
@sdurban Hello! Sorry, I've missed your topic(

It looks like you don't pass the specific destination.
Try to do it like so:
```sh
-destination "platform=iOS Simulator,name=iPhone 15"
```
```sh
set -o pipefail && env NSUnbufferedIO=YES arch -arm64 xcodebuild test \
  -workspace Example.xcworkspace \
  -scheme ExampleFrameworks \
  -testPlan ExampleFrameworks \
  -destination "platform=iOS Simulator,name=iPhone 15" \
  COMPILER_INDEX_STORE_ENABLE=NO \
  SWIFT_COMPILATION_MODE=wholemodule \
  CODE_SIGNING_ALLOWED=NO \
  | xcbeautify
```

When you don't select the specific destination `xcodebuild` tried to run it like so:
```sh
-destination generic/platform=iOS Simulator
```
And it leads to setting `ARCHS=arm64 x86_64`.
Usually, it's not we want to do. So, I've not made some fixes in 🏈 Rugby.

You are welcome!

**Reply by sdurban** [reply, marked as answer]
Thank you for your swift response!

I've implemented the destination configuration as you recommended. However, it seems that even with the specific destination set, the build process still attempts to use a combination of arm64 x86_64, which leads to the same issue persisting.

Understanding that this behavior might be unintended in many scenarios and appears to be tied to how Codemagic handles build configurations on Mac mini M1 setups, I've decided to take a temporary measure to move forward with our CI/CD processes. I'm going to fork 🏈 Rugby to apply a patch specifically addressing this issue.

This fork will allow us to proceed temporarily until I can discuss with Codemagic the reason behind the build configuration still targeting both architectures (arm64 x86_64), even after specifying a single target. I'm hopeful that this discussion will lead to a more permanent solution, whether through adjustments in Codemagic's settings or future updates to 🏈 Rugby that accommodate for this situation.

https://github.com/swiftyfinch/Rugby/compare/main...sdurban:Rugby:main

I appreciate your guidance and will update you on any progress or long-term solutions we come across!

**Reply by swiftyfinch** [reply]
Alright 👍 Also, you can check this `Build Settings` in your project:

<img width="430" alt="Screenshot 2024-03-13 at 16 09 59" src="https://github.com/swiftyfinch/Rugby/assets/64660122/e201503d-5a4e-4566-ad1c-6e758414de6b">

<img width="474" alt="Screenshot 2024-03-13 at 16 11 43" src="https://github.com/swiftyfinch/Rugby/assets/64660122/26d97eea-ae0a-42c8-a3b8-4582fe585418">

Maybe there are some issues.
